webapp/app/selectors/index.js

```diff
@@ -0,0 +1,5 @@
+export { routingSelector } from './routingSelector';
+export { usersSelector } from './usersSelector';
+export { selectedUserSelector } from './selectedUserSelector';
+export { usersLoadingSelector } from './usersLoadingSelector';
+export { usersLoadingErrorSelector } from './usersLoadingErrorSelector';
```

Comment:
I think you are polluting the selectors folder with each state you want to memoize. Imagine this folder with several modules, the best approach will be to create a usersSelector file, and there set up all selectors for the users module. Then you create a single test file. It will be easier to maintain when the app grows up. Also not all state values need to be memoized, it's better to memoize those wich triggers big screen re rendering, eg Users fetched from BE.

Reply:
We are following react conventions and scalable apps file structure for now, https://reactjs.org/docs/faq-structure.html Please consider the following example

propossed aproach here: 

`import {
    usersSelector,
    selectedUserSelector,
    fooSelector,
    barSelector
} from './selectors';

its a cleaner approach, it makes it easier and more sane to test, this adds extra visibility to what actual selectors/actions/reducers/utils we have, by simply looking at the structure, so we can spot untested files right away, as oppose to all logic in one big file.

The index file approach and small units separated in files also takes care of the following common component import issue, (which we currently have in the app)

component folder buttons with one big file buttons
`import { ActionButtons} from './components/ActionButtons/ActionButtons';`
or
`import { SOME_CONSTANT } from './contants/contants';`

This new new approach also takes care of that issue, but the main goal is Testability, Clarity and scalability as described in React docs and guides

Reply:
Please provide concrete code examples along with each comment, that way your proposed solution is cleared, thanks

Reply:
You can achieve the same with one selector file, I wrote from my experience with bigger applications in which we had dozens of selectors.

Reply:
gotcha, yes makes sense, though this initial version was implement according to redux saga's docs, ideally we will change it to follow our current actions/reducers/services approach, thanks